Solutions/DNS Domain Solution for Microsoft Sentinel/Workbooks/DNSSolutionWorkbook.json

```diff
@@ -0,0 +1,951 @@
+            },
+            "customWidth": "50",
+            "name": "query - 9",
+            "styleSettings": {
+              "showBorder": true
+            }
+          },
+          {
+            "type": 3,
+            "content": {
+              "version": "KqlItem/1.0",
+              "query": "SecurityAlert\r\n| where Entities has \"{SrcIpAddr}\" or Entities has \"{DnsQuery}\"\r\n| project TimeGenerated, DisplayName, AlertName, AlertSeverity, ProviderName, VendorName, SystemAlertId, StartTime, EndTime, todynamic(ExtendedProperties), Entities",
+              "size": 0,
+              "title": "Security Alerts with Searched IP",
+              "timeContext": {
+                "durationMs": 86400000
+              },
+              "queryType": 0,
+              "resourceType": "microsoft.operationalinsights/workspaces"
+            },
+            "customWidth": "50",
+            "name": "query - 10",
+            "styleSettings": {
+              "showBorder": true
+            }
+          },
+          {
+            "type": 3,
+            "content": {
+              "version": "KqlItem/1.0",
+              "query": "let maxSummarizedTime = toscalar (\r\n    union isfuzzy=true \r\n        (\r\n            DNS_Summarized_Logs_ip_CL\r\n                | where EventTime_t >= {TimeRange:start}\r\n                | summarize max_TimeGenerated=max(EventTime_t)\r\n                | extend max_TimeGenerated = datetime_add('hour',1,max_TimeGenerated)\r\n        ),\r\n        (\r\n            print({TimeRange:start})\r\n            | extend max_TimeGenerated = print_0\r\n            | project max_TimeGenerated\r\n        )\r\n        | summarize maxTimeGenerated = max(max_TimeGenerated) \r\n    );\r\nunion isfuzzy=true \r\n    (\r\n        _Im_Dns(starttime=todatetime(maxSummarizedTime), endtime=now())\r\n        | where ('{SrcIpAddr}' == '*' or (SrcIpAddr == '{SrcIpAddr}'))\r\n            and ('{DnsQuery}' == '*' or (DnsQuery == '{DnsQuery}'))\r\n        | where isnotempty(DnsQuery)\r\n        | extend QueryLength = strlen(DnsQuery)\r\n        | summarize TotalCount=toint(count()) by QueryLength, SrcIpAddr\r\n        | order by QueryLength, TotalCount desc\r\n        | project SrcIpAddr, QueryLength, TotalCount\r\n    ),\r\n    (\r\n        DNS_Summarized_Logs_ip_CL\r\n        | where TimeGenerated >= {TimeRange:start}\r\n        | where ('{SrcIpAddr}' == '*' or (SrcIpAddr_s == '{SrcIpAddr}'))\r\n            and ('{DnsQuery}' == '*' or (DnsQuery_s == '{DnsQuery}'))\r\n        | where isnotempty(DnsQuery_s)\r\n        | extend QueryLength = strlen(DnsQuery_s)\r\n        | summarize TotalCount=toint(sum(count__d)) by QueryLength, SrcIpAddr=SrcIpAddr_s\r\n        | order by QueryLength, TotalCount desc\r\n        | project SrcIpAddr, QueryLength, TotalCount\r\n    )\r\n    | summarize TotalCount = sum(TotalCount) by SrcIpAddr, QueryLength\r\n    | order by QueryLength, TotalCount desc\r\n    | take 25\r\n    ",
+              "size": 0,
```

Comment:
This query is resulting in points on the graph which are not adding much value. Can we switch to time series for top 5 or 10 IPs?

Reply:
Changed to top 10 instead of 25 records